Automattic/woocommerce-payments
# E2E - Add for workflow for Pull Requests, split existing workflow & introduce composite actions

Fixes #4583
Fixes #4584
Fixes #4586

#### Changes proposed in this Pull Request

This PR introduces a new E2E workflow `E2E - Pull Request` which will be run on all pull requests against `develop` or `trunk`. Also, the existing workflow is split into **3 jobs** & renamed to `E2E - Full` and will be run when commits are merged to `develop` or `trunk`, and as scheduled (every 6 hours).

Composite actions are introduced, allowing us to re-use the E2E setup scripts by referring to the composite action from any workflow, preventing code duplication.

**Here is the new workflow configuration:**

* **E2E - Pull Request**
  * Runs - on PR to `develop` or `trunk`, manual trigger
  * WooCommerce - `latest`
  * WordPress - `latest`
  * Test Groups - `wcpay`, `subscriptions`, `blocks`
  * Test Branches - `merchant`, `shopper`

* **E2E - Full**
  *  Runs - on schedule, push to `develop` or `trunk`, manual trigger 
  * **Job - `wcpay-subscriptions-tests`**
    * WooCommerce - `min`, `L-2`, `latest`, `beta`
    * WordPress - `latest`
    * Test Groups - `wcpay`, `subscriptions`
    * Test Branches - `merchant`, `shopper`
  * **Job - `blocks-tests`**
    * WooCommerce - `latest`
    * WordPress - `latest`
    * Test Groups - `blocks`
    * Test Branches - `shopper`
  *  **Job - `wp-nightly-tests`**
     * WooCommerce - `latest`
     * WordPress - `nightly`
     * Test Groups - `wcpay`, `subscriptions`, `blocks`
     * Test Branches - `merchant`, `shopper` 

Reference run - https://github.com/Automattic/woocommerce-payments/actions/runs/2955966183

**Drawbacks to this approach:**

* Previously we have used multiple steps in a workflow so we could identify the time taken at each step. However, with composite actions in place, the no of steps have reduced and it's a drawback of composite actions. 

![screenshot](https://user-images.githubusercontent.com/1161872/187439720-b129a6cb-796e-493f-9e59-66061e9e93c9.jpg)

#### Testing instructions

* Trigger tests against this branch and confirm that 3 different jobs are displayed and running as expected.
* Post merge, confirm tests are running against PRs
-------------------


**Post merge**
